gallery-so/gallery
# The POAP PR (also tezos...)

Addresses the following tickets
- [GAL-345]
- [GAL-366]
- [GAL-365]

## How to test this PR
1. Run locally
2. Point to dev
3. Login as `b_ez_man` or `kaito`
4. Check the collection editor
5. View a Gallery with POAPs and check out the NFTPreview
6. Click the community page and ensure it has `/poap` in the url
7. Check the detail page for the POAP and make sure it looks 🔥 


## Sidebar
**SPEND MOST OF YOUR TIME REVIEWING THIS**
This one underwent some serious refactoring

- Made things play much nicer with Relay
- Grouped EditModeToken and Relay Token to avoid constant map lookups (not a performance issue, just code cleanliness)
- Extracted grouping / virtualized row creation logic to separate functions for MAXIMUM PURITY
- Separated out components for my own mental health
- Introduced polling strategy for media types which are syncing (good for tezos)

## NFT Detail Page
Mostly small changes but swapped out the `Spacer` stuff for the new and improved `VStack`. This greatly helped my brain to reason about content hierarchy

## Community Page
Any link to the community page was replaced with a function to generate a url to the `/community/poap/id` vs `/community/id` based on whether or not the token is a POAP

The POAP page is a copy paste of the other page with a different chain.

## TODO
- [x] Add empty states for the chain tabs in the collection editor
- [x] Fix Robin's bug
  - This is actually a really annoying bug with the multiline extra spacing. Can discuss in GM
- [x] Wizard refreshing thing
- [x] More Info / View on POAP
  - Need to include POAP ID in the token metadata

## Open Questions

How do we want to land this? 
- Feature flag?
- Custom hotkey to enable?
- Allow list?
